MetaMask/metamask-extension
# Show custom tokens in Swaps, add a custom token in Swaps

## Explanation
This PR solves a few things:
- Shows custom tokens in Swaps if a user added them on the main page in the `Assets` tab
- Adds a possibility to paste a contract address in Swaps and if a token is found, a user can import it
- Fixes a bug where we were showing a scrollbar in a dropdown for tokens, even if there was only 1 token
- Fixes a bug where we were pushing away a token icon in the `Swap From` dropdown if a token name was too long
- Removes a cursor from a heading

TODOs:
- Test a successful swap with a custom token added on the main page
- Test a successful swap with a custom token added in Swaps

## Manual testing steps
Show a custom token in Swaps:
- On the main page go into the `Assets` tab and click on the `Add Token` button
- Click on `Custom Token`
- Paste your contract address and click on `Next`
- Click on `Swap` on the main page
- Search for your custom token on the `Build Quote` page, you should see your custom token there

Add a custom token in Swaps:
- Click on `Swap` on the main page
- Click on `Select a token` for the `Select to` dropdown
- Paste a contract address of your custom token
- If a token is found, you will see one item in the list and it will have an `Import` button
- Click on the item in the list or its `Import` button
- A new modal window for importing a token is displayed
- Click on `Import`
- Your custom token is selected in the `Select to` dropdown
 
## Screenshots
Show a notification with a list of tokens about adding a token via a contract address:
<img width="355" alt="image" src="https://user-images.githubusercontent.com/80175477/120196680-6853da80-c220-11eb-9da2-04bf9a766fe5.png">

Show a notification if no token is found about adding a token via a contract address:
<img width="354" alt="image" src="https://user-images.githubusercontent.com/80175477/120196704-70ac1580-c220-11eb-8958-56282c8e75a8.png">

Token found by a contract address:
<img width="350" alt="image" src="https://user-images.githubusercontent.com/80175477/119868779-c7afa300-bf1f-11eb-927a-809c3db19af4.png">

Import token popup:
<img width="357" alt="image" src="https://user-images.githubusercontent.com/80175477/120196745-7efa3180-c220-11eb-9e36-0ddfe43993a0.png">

Token imported:
<img width="351" alt="image" src="https://user-images.githubusercontent.com/80175477/119868856-de55fa00-bf1f-11eb-89f6-9ffca4a77f27.png">

List of custom tokens in the `Assets` tab on the main page:
<img width="357" alt="image" src="https://user-images.githubusercontent.com/80175477/119976707-cc279a80-bfb7-11eb-8c5a-bd8f548aa075.png">

List of custom tokens when selecting tokens in Swaps:
<img width="357" alt="image" src="https://user-images.githubusercontent.com/80175477/119976778-e792a580-bfb7-11eb-90b5-297937ee9d27.png">
